### PR TITLE
PHPstan level 6 for I18nFile.php

### PR DESCRIPTION
**Pull request by ColonelMoutarde**
Changes proposed in this pull request:

- PHPstan level 6 for I18nFile.php 

Pull request checklist:
- [x] clear commit messages
- [x] code manually tested

### CODE REVIEW

**Comment by Alkarex** [comment]
Will check tonight 👍🏻 
(FYI, I have a work in progress on FreshRSS_EntryDAO and FreshRSS_Search, so you do not start these two, @ColonelMoutarde )

File: cli/i18n/I18nFile.php [hunk 1]
```diff
@@ -23,7 +26,10 @@ public function load() {
 		return $i18n;
 	}
 
-	public function dump(array $i18n) {
+	/**
+	 * @param array<string,array<array<string>>> $i18n
```
Comment (Alkarex) [inline cli/i18n/I18nFile.php:30]: I am not sure of the type. It needs to work with:

https://github.com/FreshRSS/FreshRSS/blob/6e7fa07a39e2ff73c3e5e11d38a61d1f2231c859/cli/manipulate.translation.php#L83

Could you please try to do the types for `I18nData`? I believe this is needed to get the types here right

Comment (ColonelMoutarde) [reply cli/i18n/I18nFile.php:30]: ok

Comment (ColonelMoutarde) [reply cli/i18n/I18nFile.php:30]: done

File: cli/i18n/I18nData.php [hunk 2]
```diff
@@ -14,20 +14,25 @@ public function __construct(array $data) {
...
 		return $this->data;
 	}
 
-	private function addMissingKeysFromReference() {
+	private function addMissingKeysFromReference(): void {
 		$reference = $this->getReferenceLanguage();
 		$languages = $this->getNonReferenceLanguages();
 
 		foreach ($reference as $file => $refValues) {
+			/** @var I18nValue $refValue */
```
Comment (Alkarex) [inline cli/i18n/I18nData.php]: It should not be necessary to manually override the types there. It is likely a sign of problem for the signature of `$this->getReferenceLanguage()`

File: cli/i18n/I18nData.php [hunk 3]
```diff
@@ -318,36 +306,35 @@ public function ignore($key, $language, $reverse = false) {
...
-	 * @param boolean $reverse
 	 */
-	public function ignore_unmodified($language, $reverse = false) {
+	public function ignore_unmodified(string $language, bool $reverse = false): void {
 		$my_language = $this->getLanguage($language);
 		foreach ($this->getReferenceLanguage() as $file => $ref_language) {
 			foreach ($ref_language as $key => $ref_value) {
 				if (array_key_exists($key, $my_language[$file])) {
-					if($ref_value->equal($my_language[$file][$key])) {
+					if ($ref_value->equal($my_language[$file][$key])) {
```
Comment (Alkarex) [inline cli/i18n/I18nData.php:309]: `$ref_value` does not look typed here. Again a sign that the signature of `getReferenceLanguage()` might not be precise enough.
Intelephense gives a warning as well:

```
Expected type 'object'. Found 'string|string[]'.
```

**Comment by Alkarex** [comment]
I have fixed several type errors https://github.com/FreshRSS/FreshRSS/pull/5291/commits/3705dd3a6bc2c0392bd865e15e66cbd82b171775
Contributes to https://github.com/FreshRSS/FreshRSS/issues/4112

**Pull request merged**